ethereum/spec/src/main/java/tech/pegasys/teku/spec/Spec.java

```diff
@@ -1118,21 +1146,17 @@ public int hashCode() {
 
   public Optional<Integer> getMaxBlobsPerBlockAtSlot(final UInt64 slot) {
     final SpecVersion specVersion = atSlot(slot);
-    switch (specVersion.getMilestone()) {
-      case PHASE0, ALTAIR, BELLATRIX, CAPELLA -> {
-        return Optional.empty();
-      }
-      case DENEB, ELECTRA -> {
-        return Optional.of(
-            specVersion.getConfig().toVersionDeneb().orElseThrow().getMaxBlobsPerBlock());
-      }
-      default -> {
-        final UInt64 epoch = atSlot(slot).miscHelpers().computeEpochAtSlot(slot);
-        return Optional.of(
+    return switch (specVersion.getMilestone()) {
+      case PHASE0, ALTAIR, BELLATRIX, CAPELLA -> Optional.empty();
+      case DENEB, ELECTRA ->
+          Optional.of(SpecConfigDeneb.required(specVersion.getConfig()).getMaxBlobsPerBlock());
+      case FULU -> {
+        final UInt64 epoch = specVersion.miscHelpers().computeEpochAtSlot(slot);
+        yield Optional.of(
```

Comment:
this would mean we need to build out this every single upgrade... i think the previous version is probably cleaner.